pkg/controller/instance/instance_controller.go

```diff
@@ -199,6 +235,13 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 
 	if instance.Status.AggregatedStatus.Status.IsTerminal() {
 		r.Recorder.Event(instance, "Normal", "PlanFinished", fmt.Sprintf("Execution of plan %s finished with status %s", activePlanStatus.Name, instance.Status.AggregatedStatus.Status))
+
+		lastExecutedPlanStatus := instance.GetLastExecutedPlanStatus()
+		if lastExecutedPlanStatus != nil && lastExecutedPlanStatus.Name == kudov1beta1.CleanupPlanName {
```

Comment:
I would love if this already happened earlier, so we can update the instance as part of `updateInstance` method on line 230, that way we're not updating instance on many places, just that one...